core/domain/exp_services.py

```diff
@@ -927,9 +927,11 @@ def update_states_version_history(
             # Deleting the attributes from the state dicts which are present
             # in the ignore list.
             for property_name in state_property_ignore_list:
-                # MyPy doesn't allow key deletion from TypedDict,
-                # thus we add an ignore.
+                # Here we use MyPy ignore because MyPy doesn't allow key
```

Comment:
I'm not sure what do mean by this -- because if you see the usage there we are deleting the key dynamically. Also ( again ) no matter if we define any kind of TypeDDict, if we are deleting the key then MyPy will throw an error.

Reply:
Removed new_state_dict and old_state_dict